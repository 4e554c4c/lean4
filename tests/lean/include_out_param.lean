class c (α : Type) (β : out_param Type) :=
(n : α → nat)

variables {α β : Type} [c α β]

def f : nat := 1
#print f -- don't include anything
def g (a : α) : nat := c.n a
#print g -- include everything
