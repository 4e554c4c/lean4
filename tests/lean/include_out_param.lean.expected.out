def f : ℕ :=
1
def g : Π {α β : Type} [_inst_1 : c α β], α → ℕ :=
λ {α β : Type} [_inst_1 : c α β] (a : α), c.n a
